blocks/snippets.py

```diff
@@ -0,0 +1,14 @@
+"""Routines to build frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    return tensor.sqrt(sum((t ** 2).sum() for t in tensors))
```

Comment:
But they're not like snippets, at all! The only thing that distinguishes snippets from helper-functions/macros is that the former are per definition copy-pasted into your code (most of the time by your IDE). Using that argument, a snippet and a helper-function are the same thing, which they're not.

I think the best alternative is `expressions`, which describes exactly what they are: factored out Theano expressions Yes, strictly speaking they are functions that *return* expressions, but we name functions and files all the time after things that they return/provide; I've never seen someone call a file "fibonacci_returnes" instead of "fibonacci". I've recently already replaced a number of "Theano expressions" by "Theano variables", so I'd rather finish up doing that if you really think the dual use of the word "expressions" is going to confuse people.

Alternative names would be "theano_macros", "theano_utils", where the latter has my preference (because macros, like snippets, have a specific use that is not the case here).